pinot-common/src/test/java/com/linkedin/pinot/common/data/SchemaTest.java

```diff
@@ -15,97 +15,105 @@
  */
 package com.linkedin.pinot.common.data;
 
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
 import java.util.concurrent.TimeUnit;
-import org.codehaus.jackson.map.ObjectMapper;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
 import org.testng.annotations.Test;
 
 
 public class SchemaTest {
   public static final Logger LOGGER = LoggerFactory.getLogger(SchemaTest.class);
 
-  private static String singleValueDim = "true";
-  private static String dimType = "\"STRING\"";
-  private static String metricType = "\"LONG\"";
+  private Schema schema;
 
-  private String makeSchema() {
+  @BeforeClass
+  public void setUp()
+      throws IOException {
+    URL resourceUrl = getClass().getClassLoader().getResource("schemaTest.schema");
+    Assert.assertNotNull(resourceUrl);
+    schema = Schema.fromFile(new File(resourceUrl.getFile()));
+  }
+
+  @Test
+  public void testValidation() throws Exception {
+    Schema schemaToValidate;
+
```

Comment:
Currently, the Schema validate() only check the data type and default null value for all fields.